### ISSUE TITLE
`Layout/ExtraSpacing` strange positive

### ISSUE THREAD

**Issue by deivid-rodriguez** [rubocop/rubocop#7824]
`Layout/ExtraSpacing` is registering an offense that I wasn't expecting.

## Expected behavior

I have the following array:

#### source.rb

```ruby
expected = [
  [:section, 'GEM',                0, 0],
  [:newline, nil,                  3, 0],
  [:entry,   'remote',             2, 1],
  [:text,    @gem_repo,           10, 1],
  [:newline, nil,                 34, 1],
  [:entry,   'specs',              2, 2],
  [:newline, nil,                  8, 2],
  [:text,    'Ab',                 4, 3],
  [:l_paren, nil,                  7, 3],
  [:text,    '2',                  8, 3],
  [:r_paren, nil,                  9, 3],
  [:newline, nil,                 10, 3],
  [:newline, nil,                  0, 4],
  [:section, 'PLATFORMS',          0, 5],
  [:newline, nil,                  9, 5],
  [:text,    Gem::Platform::RUBY,  2, 6],
  [:newline, nil,                  6, 6],
  [:newline, nil,                  0, 7],
  [:section, 'DEPENDENCIES',       0, 8],
  [:text,    'Ab',                 2, 9],
  [:newline, nil,                  4, 9],
]
```

I'd expect `Layout/ExtraSpacing` to not register any offenses, because extra spacing is allowed for alignment by default.

## Actual behavior

An offense is registered exclusively for line 13:

```
$ rubocop source.rb 
Inspecting 1 file
C

Offenses:

source.rb:13:18: C: Layout/ExtraSpacing: Unnecessary spacing detected.
  [:newline, nil,                 10, 3],
                 ^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

Create a ruby file with that array and run rubocop on it.

## RuboCop version

```
$ [bundle exec] rubocop -V
0.80.1 (using Parser 2.7.0.5, running on ruby 2.7.0 x86_64-linux)
```

**Comment by deivid-rodriguez** [comment]
I just noticed that my array is not consistently aligned (second column is left-aligned, third column is right-aligned), so this is probably not an issue (although I'm still surprised that only the 13th line registers an offense?). Left-aligning the third column fixes the issue.

**Comment by dmytro-savochkin** [comment]
This happens because this cop checks alignment of a value in a column with any of two adjacent lines.
So, suppose we have this code:
```
[
  [:section, 'GEM',                0, 0],
  [:entry,   'remote',             2, 1],
  [:text,    @gem_repo,           10, 1],
  [:newline, nil,                 34, 1],
  [:entry,   'specs',              2, 2],
  [:r_paren, nil,                  9, 3],
  [:newline, nil,                 10, 3],
  [:newline, nil,                  0, 4]
]
```
Here line 1 is aligned with line 2, line 2 is aligned with line 1, line 3 is aligned with line 4, line 4 is aligned with line 3, line 5 is aligned with line 6, line 6 is aligned with line 5, but then line 7 is not aligned neither with line 6 nor line 8 hence an offense.

At a first glance seems pretty hard to fix to be honest.

**Comment by dvandersluis** [comment]
It's been two years since the last update, and like was said above, there isn't an easy fix for this. If anyone affected by this issue wants to open a PR, please create one!
